torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -710,7 +710,22 @@ def triton_kernel_wrapper_mutation_dense(
                 element_size,
             )
 
-    kernel[grid_fn](**kwargs, **constant_args)
+    # move as many positional arguments from dicts to args as we
+    # can to circumvent the bug with the kwargs and pre_/post_hook:
+    # https://github.com/triton-lang/triton/issues/5082
+    # TODO: remove this when the Triton issue above is fixed
+    args = []
```

Comment:
this feels really bad, and confusing. Maybe only do this when pre/post hook is used?

Reply:
At this stage of the Triton code, we can't tell when pre/post hook is used. Just submitted a PR so that we'll be able to tell. Can update then in a follow-up: https://github.com/triton-lang/triton/pull/5092. Although, there is another PR fixing the mentioned issue that should make this whole thing unnecessary: https://github.com/triton-lang/triton/pull/5083.

Reply:
Could you elaborate why it's bad, btw? Is this incorrect? I'm just moving the args around, but not skipping or duplicating any passed to the kernel?